### DISCUSSION TITLE
There is a way to change context type based on the auth scopes?

### DISCUSSION THREAD

**Discussion by willianhf** [hayes/pothos#350] [Q&A] [answered]
I want to try to change some context typings based on the auth scoped passed to a field.

For example:
```ts
type SchemaBuilderConfig = {
  Context: {
    userAgent: string;
    user: User | null;
  };
  AuthScopes: {
    isLoggedIn: boolean;
    isAdmin: boolean;
  };
}

export const schemaBuilder = new SchemaBuilder<SchemaBuilderConfig>({
  plugins: [ScopeAuthPlugin],
  authScopes: async context => ({
    isLoggedIn: !!context.user,
    isAdmin: !!context.user?.isAdmin
  }),
  scopeAuthOptions: {...}
});

schemaBuilder.queryType({
  fields: t => ({
    hello: t.string({
      authScopes: {
        isLoggedIn: true
      },
      resolve: (_root, _args, context) => {
        // context.user type to be User since isLoggedIn is required
        return 'Hello world!';
      }
    })
  })
});
```

For while I'm using the non-null assertion operator.

**Comment by hayes** [comment, marked as answer]
Seems like something got dropped from the docs 


https://github.com/hayes/pothos/blob/7dd293dbb5e58412cec81fb78d38596384c52e64/packages/plugin-scope-auth/tests/example/schema/index.ts#L5

https://github.com/hayes/pothos/blob/7dd293dbb5e58412cec81fb78d38596384c52e64/packages/plugin-scope-auth/tests/example/builder.ts#L20

Basically, you can provide custom contexts for your scopes and use auth field method to get a custom context

**Reply by willianhf** [reply]
That is exactly what I need, thanks for the quick answer!
Should I open a PR to add it back to the docs?

**Reply by hayes** [reply]
Sure, that would be great!
